src/antsibull_core/collections.py

```diff
@@ -9,35 +9,42 @@
 
 import asyncio
 import os
+import typing as t
 
 from .subprocess_util import async_log_run
 
+if t.TYPE_CHECKING:
+    from _typeshed import StrPath
+
 
 class CollectionFormatError(Exception):
     pass
 
 
 async def install_together(
-    collection_tarballs: list[str], ansible_collections_dir: str
+    collection_tarballs: list[StrPath], ansible_collections_dir: StrPath
 ) -> None:
     installers = []
     for pathname in collection_tarballs:
-        namespace, collection, _dummy = os.path.basename(pathname).split("-", 2)
+        basename = os.path.basename(
+            pathname  # pyre-ignore[6]: basename() accepts path-like object
+        )
+        namespace, collection, _dummy = basename.split("-", 2)
         collection_dir = os.path.join(ansible_collections_dir, namespace, collection)
         # Note: mkdir -p equivalent is okay because we created package_dir ourselves as a directory
         # that only we can access
         os.makedirs(collection_dir, mode=0o700, exist_ok=False)
         installers.append(
             asyncio.create_task(
-                async_log_run(["tar", "-xf", pathname, "-C", collection_dir])
+                async_log_run(["tar", "-xf", f"{pathname}", "-C", collection_dir])
```

Comment:
It shouldn't be necessary to stringify `pathname`. `async_log_run()` can already handle this.

Reply:
44445f2fa57956dfcb84535fbfc21edf0efe6fc0.